src/client/lib/hooks/useCmpConsent.ts

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react';
+import { ConsentState } from '@guardian/consent-management-platform/dist/types';
+import { onConsentChange } from '@guardian/consent-management-platform';
+
+/**
+ * ConsentState.canTarget checks if user has consented to all consents
+ * https://github.com/guardian/consent-management-platform#consentstatecantarget
+ */
+const hasRequiredCmpConsents = (): Promise<boolean> => {
+  return new Promise((resolve) => {
```

Comment:
Just fyi if you don't need to listen for subsequent consent changes then the new method `onConsent` method will give you a promise with the consent state:
https://github.com/guardian/consent-management-platform#onconsent

Reply:
Thanks. I think onConsent will work for us.